### PR TITLE
storage: revert internal control flow changes

### PR DESCRIPTION
**Pull request by aljoscha**
### Motivation

It seems that the work of introducing the internal control flow for getting rid of `halt!` caused https://github.com/MaterializeInc/materialize/issues/17309. 

### Tips for reviewer

This is tricky because things got piled on top. We need to check carefully.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a
  companion cloud PR to account for those changes that is tagged with
  the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  -

### CODE REVIEW

File: src/storage/src/server.rs [hunk 1]
```diff
@@ -107,7 +107,8 @@ pub fn serve(
                 dropped_ids: Vec::new(),
                 source_statistics: HashMap::new(),
                 sink_statistics: HashMap::new(),
-                internal_cmd_tx: None,
+                // TODO(aljoscha): We can refactor when `StorageState` is being
+                // created, to get around this being an `Option`.
```
Comment (guswynn) [inline src/storage/src/server.rs:111]: this can be deleted, but i would rather merge this asap

**Review by guswynn — APPROVED** [review]
This seems reasonable to me!

Can we squash merge so its easier to revert?

**Comment by guswynn** [comment]
Also, i started a release qualification build to see if the same issues come up: https://buildkite.com/materialize/release-qualification/builds/70

**Comment by aljoscha** [comment]
I'm fine with squashing, yes!

**Comment by guswynn** [comment]
Nightly `UserTables` and the RQ test have passed, so im squashing

**Pull request merged**